lib/Elastica/Filter/AbstractFilter.php

```diff
@@ -4,12 +4,16 @@
 use Elastica\Exception\InvalidException;
 use Elastica\Param;
 
+trigger_error('Deprecated: Filters are deprecated. Use queries in filter context. See https://www.elastic.co/guide/en/elasticsearch/reference/2.0/query-dsl-filters.html', E_USER_DEPRECATED);
+
 /**
  * Abstract filter object. Should be extended by all filter types.
  *
  * @author Nicolas Ruflin <spam@ruflin.com>
  *
  * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-filters.html
+ *
+ * @deprecated Filters are deprecated. Use queries in filter context. See https://www.elastic.co/guide/en/elasticsearch/reference/2.0/query-dsl-filters.html
  */
 abstract class AbstractFilter extends Param
```

Comment:
My initial thought was, that one extends the other which would prevent the check function above as we could still have the type specified in the function. Still thinking if this could make sense? That would remove the throw InvalidException part.